### PR TITLE
RR-475 display goal updated event on the timeline

### PR DESCRIPTION
**Pull request by chrimesdev**
## Description

Display "Goal updated" event on the Timeline UI.

Also add some tests for the Timeline view and Timeline service.

https://dsdmoj.atlassian.net/browse/RR-475

### Screenshots

screencapture-localhost-3000-plan-G5045UF-view-timeline-2023-11-21-16_00_00

screencapture-localhost-3000-plan-A9412DY-view-timeline-2023-11-21-15_54_58

### CODE REVIEW

File: server/views/pages/overview/partials/timelineTab/timelineTabContents.njk [hunk 1]
```diff
@@ -7,17 +7,24 @@
...
 
           <p class="moj-timeline__date">
             <time datetime="{{ event.timestamp }}">{{ event.timestamp | formatDate('D MMMM YYYY') }}</time>
           </p>
 
-          {% if event.eventType == 'ACTION_PLAN_CREATED' %}
+          {% if event.eventType == 'ACTION_PLAN_CREATED' or event.eventType == 'GOAL_UPDATED' %}
           <div class="moj-timeline__description">
-            <p class="govuk-body govuk-!-margin-bottom-0 govuk-!-display-none-print"><a href="/plan/{{ prisonerSummary.prisonNumber }}/view/overview">View {{ prisonerSummary.firstName }} {{ prisonerSummary.lastName | title }}'s learning and work progress</a></p>
+            <p class="govuk-body govuk-!-margin-bottom-0 govuk-!-display-none-print">
```
Comment (nathanrussell-moj-digital) [inline server/views/pages/overview/partials/timelineTab/timelineTabContents.njk:20]: This is OK for now, but once we add a 3rd type that has subtly different link/display requirements here this is going to get messy! 😢 

I wonder if the next time we need to touch this section of the template we look to do it a different way, perhaps with a conditional or "dynamic" include for each event type? 🤔 

As mentioned, its OK for now, but next time we need to touch this lets look at alternative ways of doing it 👍

File: server/services/timelineService.ts [hunk 2]
```diff
@@ -19,10 +19,12 @@ export default class TimelineService {
     }
   }
 
-  filterTimelineByActionPlanCreatedOrUpdatedEvent = (timelineResponse: TimelineResponse) => {
+  filterTimelineByActionPlanOrGoalEvent = (timelineResponse: TimelineResponse) => {
```
Comment (nathanrussell-moj-digital) [inline server/services/timelineService.ts:22]: My PR will cause a merge conflict here.
I think what you'll want to do is take the version of this file `timelineService.ts` from `main`, and simply add `GOAL_UPDATED` to the array `SUPPORTED_TIMELINE_EVENTS`

Comment (chrimesdev) [reply server/services/timelineService.ts:22]: Sounds good - I can resolve the conflict once your PR is merged

File: server/views/pages/overview/partials/timelineTab/timelineTabContents.njk (hunk 1 above)
Comment (chrimesdev) [reply server/views/pages/overview/partials/timelineTab/timelineTabContents.njk:20]: Yep makes sense, I think the "Goal created" event has the same link as "Goal updated" so we may be ok for a little longer - but any more differences then definitely.

File: server/routes/overview/overviewController.test.ts [hunk 3]
```diff
@@ -476,4 +477,42 @@ describe('overviewController', () => {
       expect(ciagInductionService.getWorkAndInterests).toHaveBeenCalledWith(prisonNumber, 'a-user-token')
     })
   })
+
+  describe('getTimelineView', () => {
```
Comment (nathanrussell-moj-digital) [inline server/routes/overview/overviewController.test.ts:481]: 👍

**Pull request merged**